masschallenge/django-accelerator
# AC-6391: Construct and implement the subnav menu typology using django-sitetree and semantic ui

#### Changes introduced in [AC-6391](https://masschallenge.atlassian.net/browse/AC-6391)
- add migration to create the startup dashboard subnav

#### How to test
#### this setup is deployed on test3
- masquerade as [super+asante](https://test3.masschallenge.org/admin/simpleuser/user/60729/masquerade/) who I have given Finalist ProgramRoleGrants to both the Fintech and HealthTech programs
- see the [HealthTech calendar page](https://test3.masschallenge.org/dashboard/ht-finalist/calendar/) and notice the menu populated with both the HT and FT calendar links
- also the [HealthTech Dashboard](https://test3.masschallenge.org/dashboard/ht-finalist/landing-page/)  should show 2 menu items from both programs

#### if you prefer to test on your local environment

- checkout this branch on accelerate and accelerator
- if you have been testing other PRs and running migrations I would recommend you `make load-db` to ensure you start a fresh
- `make django-shell` and set up a user and permissions as follows
```
from mc.models import *
user = User.objects.get(email__startswith="demoadmin")
Clearance.objects.create(user=user, level=CLEARANCE_LEVEL_GLOBAL_MANAGER, program_family=ProgramFamily.objects.first())
user.is_staff = user.is_superuser = True
user.set_password('secret')
user.save()

program = Program.objects.filter(program_status='active')
role = UserRole.objects.filter(name="Finalist").first()
prs = ProgramRole.objects.filter(user_role=role, program__in=program)
for pr in prs:
    ProgramRoleGrant.objects.create(person=user, program_role=pr)
```

- now log in with this super powerful user you just created
- visit the PULSE home [dashboard page](http://localhost:8181/dashboard/pulse-cohort/
pulse-dashboard-home/) and notice we have no sub navigation showing up
- visit the pulse [home dashboard on the admin](http://localhost:8181/admin/fluent_pages/page/616/change/)
- now in the "FLUENT PAGE LINK WILL SHOW UP ON" section, associate this page with
    - side_nav: "home"subnav"
    - sub nav item: "Finalist Dashboard"
- and "Save"
- now visit the PULSE home [dashboard page](http://localhost:8181/dashboard/pulse-cohort/ again and notice a sub navigation item added
#### Note
- for these menu items to show up on the fluent sub nav, I had to create an association between a fluent page and the sub navigation, e.g. for the [HealthTech Program Calendar](https://test3.masschallenge.org/admin/fluent_pages/page/613/change/), you will notice it was associated with the `events_subnav` and the `Calendar` sub nav item.

<img width="1228" alt="Screen Shot 2019-04-01 at 13 55 45" src="https://user-images.githubusercontent.com/9318050/55322756-e1f7ae80-5485-11e9-853c-f98260b13653.png">

Associating a fluent page with 2 sub navigation trees will default to the first association when displaying a fluent page sub navigation menu.
